// Copyright 2023 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "src/proto/grpc/service_config/service_config.proto";

package service_config_fuzzer;

message ServiceConfigChoice {
    repeated string client_languages = 1;
    int32 percentage = 2;
    repeated string client_hostnames = 3;
    grpc.service_config.ServiceConfig service_config = 4;
}

message ServiceConfigTXTRecord {
    repeated ServiceConfigChoice service_config_choices = 1;
}

message Msg {
    oneof payload {
        ServiceConfigTXTRecord service_config_txt_record = 1;
        string arbitrary_txt_record = 2;
    }
}
